src/pages/oncall/oncall.ts

```diff
@@ -46,31 +46,41 @@ export class OncallPage {
 
   initOncallLists(){
     
-    this.iris.clearOncallUsers;
-        this.iris.getOncallUsers().subscribe(
-          () => {
-            this.oncallUsers = this.iris.oncallUsers;
-            this.oncallUsers = this.oncallUsers.sort((a, b) => {if(b.name > a.name){return -1;}else{return 1;}})
-            this.unfilteredUsers = [];
-            for (let i = 0; i < this.oncallUsers.length; i++) {
-              this.unfilteredUsers.push({username: this.oncallUsers[i].name, full_name: this.oncallUsers[i].full_name});
-            }
-            this.usersLoading = false;
-          },
-          () => {
-            this.createToast('Error: failed to fetch oncall users.');
-            this.loadingError = true;
-          },
+    // if users are saved in storage initialize list with that data while we wait from response from server
+    if(this.iris.oncallUsersLoaded){
+      this.oncallUsers = this.iris.oncallUsers;
+      this.oncallUsers = this.oncallUsers.sort((a, b) => {if(b.name > a.name){return -1;}else{return 1;}})
+      this.unfilteredUsers = [];
+      for (let i = 0; i < this.oncallUsers.length; i++) {
+        this.unfilteredUsers.push({username: this.oncallUsers[i].name, full_name: this.oncallUsers[i].full_name});
+      }
+      this.usersLoading = false;
+    }
+
+    this.iris.getOncallUsers().subscribe(
+      () => {
+        this.oncallUsers = this.iris.oncallUsers;
+        this.oncallUsers = this.oncallUsers.sort((a, b) => {if(b.name > a.name){return -1;}else{return 1;}})
+        this.unfilteredUsers = [];
+        for (let i = 0; i < this.oncallUsers.length; i++) {
+          this.unfilteredUsers.push({username: this.oncallUsers[i].name, full_name: this.oncallUsers[i].full_name});
```

Comment:
Any reason for an Object here, rather than the `OncallUser` type?

Reply:
It's a holdover from when the code was structured differently, I'll get rid of it since it is redundant now.